hermes/bin/src/runtime_extensions/hermes/cardano/event.rs

```diff
@@ -1,20 +1,29 @@
 //! Cardano Blockchain runtime extension event handler implementation.
 
 use crate::{
-    event::HermesEventPayload,
-    runtime_extensions::bindings::hermes::cardano::api::{
-        BlockSrc, CardanoBlock, CardanoBlockchainId, CardanoTxn,
+    app::ApplicationName,
+    event::{HermesEvent, HermesEventPayload, TargetApp, TargetModule},
+    runtime_extensions::bindings::{
+        exports::hermes::cardano::event_on_block::{CardanoNetwork, SubscriptionId},
+        hermes::cardano::api::Slot,
     },
+    wasm::module::ModuleId,
 };
 
 /// On Cardano block event
 pub(super) struct OnCardanoBlockEvent {
-    /// The blockchain id the block originated from.
-    pub(super) blockchain: CardanoBlockchainId,
-    /// This raw CBOR block data.
-    pub(super) block: CardanoBlock,
-    /// Source information about where the block came from, and if we are at tip or not.
-    pub(super) source: BlockSrc,
+    /// The Cardano blockchain network that the block event is originated from.
+    network: CardanoNetwork,
+    /// A unique identifier of the block subscription.
+    subscription_id: SubscriptionId,
+    /// The Cardano slot number that the block event is in.
+    slot: Slot,
+    /// The CBOR format of the Cardano block data.
+    block: Vec<u8>,
+    /// Flag indicate whether the block is immutable or not.
+    is_immutable: bool,
+    /// Flag indicate whether the block is a rollback.
+    is_rollback: bool,
```

Comment:
With the above two changes, the event would just be:
```
pub(super) struct OnCardanoBlockEvent {
    subscription_id: SubscriptionId,
    block: Block
}
```
which looks a lot cleaner / easier to manage / maintain.